resources/assets/components/Signup/index.js

```diff
@@ -184,6 +211,46 @@ class Signup extends React.Component {
     }
   }
 
+  // Submit a new reportback on behalf of the user.
+  submitReportback(reportback) {
+    // Fields to send to /posts
+    const fields = {
+      northstar_id: reportback.northstarId,
+      campaign_id: reportback.campaignId,
+      campaign_run_id: reportback.campaignRunId,
+      quantity: reportback.impact,
+      why_participated: reportback.whyParticipated,
+      caption: reportback.caption,
+      source: reportback.source,
+      status: reportback.status,
+      file: reportback.media.dataURL,
+      remote_addr: '',
+    };
+
+    // Make API request to Rogue to upload post
+    let request = this.api.post('posts', fields);
+
+    request.then((result) => {
+      // Update the state
+      this.setState((previousState) => {
+        const newState = {...previousState};
+        const post = keyBy(result,'id');
+        const key = Number(Object.keys(post));
+
+        newState.successfulSubmission = {
+          success: {
+            message: "Thanks for the photo! It has been automatically approved.",
+          }
+        };
+
+        newState.posts[key] = post[key];
+
+        return newState;
+      });
+    })
+    .catch(error => console.log(error));
```

Comment:
Instead of `console.log(error)`, should I set an error message? Something like the below?
```
newState.successfulSubmission = {
  error: {
    message: "Something went wrong.",
  }
};
```

Reply:
I think the error message is a good idea! I'm not too concerned with the success message staying up there, unless we here otherwise

Reply:
ok cool!